src/helpers/workspace.ts

```diff
@@ -185,3 +186,25 @@ async function selectTruffleConfigFromQuickPick(workspaces: TruffleWorkspace[]):
 
   return result.truffleWorkspace;
 }
+
+/**
+ * Every time the `workspace.onDidSaveTextDocument` listener emits a notification,
+ * this function receives, identifies the file extension and calls the corresponding function.
+ *
+ * @param globalState A memento object that stores state independent of the current opened workspace.
+ * @param document Represents a text document, such as a source file.
+ */
+export async function saveTextDocument(globalState: Memento, document: TextDocument): Promise<void> {
+  switch (path.extname(document.fileName)) {
+    case '.sol': {
+      // Gets the current state of the status bar item
+      const isAutoDeployOnSaveEnabled = globalState.get<boolean>(Constants.globalStateKeys.contractAutoDeployOnSave);
+
+      // If enabled, calls the function that performs the deployment
+      if (isAutoDeployOnSaveEnabled) await TruffleCommands.deployContracts(Uri.parse(document.fileName));
```

Comment:
It would be neat to (optionally) persist the selected destination so it doesn't require the user to select each time. No need for now though, we can consider for a future sprint.